### PR TITLE
[Merged by Bors] - Update ID calculation of Proposal

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
With the updated method to calculate ATX IDs the whole object including the signature is used for ID calculation. This PR updates the ID calculation of Proposals to align with this new method. See also https://github.com/spacemeshos/go-spacemesh/issues/3516

## Changes
Update method to calculate Proposal ID calculation.

## Test Plan
Existing tests still pass.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by fasmat** [comment]
bors try

**Comment by fasmat** [comment]
bors try

**Review by dshulyak — APPROVED** [review]

File: common/types/hashes.go [hunk 1]
```diff
@@ -177,7 +177,16 @@ func CalcHash32(data []byte) Hash32 {
...
 	bytes, err := codec.Encode(atx)
 	if err != nil {
-		panic("could not Serialize atx")
+		panic("could not serialize ATX")
+	}
+	return CalcHash32(bytes)
+}
+
+// CalcProposalHash32 returns the 32-byte sha256 sum of serialization of the given Proposal.
+func CalcProposalHash32(proposal *Proposal) Hash32 {
```
Comment (dshulyak) [inline common/types/hashes.go]: it looks to me like there could be one generic function to compute hash from scale.Encodable

Comment (fasmat) [reply common/types/hashes.go]: Created a function that takes a `scale.Encodable` for `Hash32` calculation of an object.

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**